### PR TITLE
[review] Fix uninitialized members

### PR DESCRIPTION
**Pull request by dawagner**
Some data members were left uninitialized in some classes.

Use default-member-initializers in all libparameter to help avoid such mistake in the future and fix the current issues.
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/01org/parameter-framework/pull/296%23issuecomment-151420999%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/296%23issuecomment-151878954%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/296%23issuecomment-152514334%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/296%23issuecomment-152969640%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/296%23discussion_r43099858%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/296%23discussion_r43099933%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/296%23discussion_r43106257%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/296%23discussion_r43106268%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/296%23discussion_r43118851%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/296%23discussion_r43118854%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/296%23discussion_r43128084%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/296%23discussion_r43130014%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/296%23discussion_r43130153%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/296%23discussion_r43609768%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/296%23issuecomment-151420999%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%40krocard%20%40tcahuzax%20please%20review.%22%2C%20%22created_at%22%3A%20%222015-10-27T09%3A07%3A45Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40tcahuzax%20Please%20review.%22%2C%20%22created_at%22%3A%20%222015-10-28T15%3A19%3A10Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A-1%3A%22%2C%20%22created_at%22%3A%20%222015-10-30T12%3A48%3A05Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/11178583%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tcahuzax%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%20%22%2C%20%22created_at%22%3A%20%222015-11-02T09%3A32%3A22Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%202c8eda08af99554bf66de2148f1696324b1504c8%20parameter/ConfigurableElement.cpp%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/296%23discussion_r43128084%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22default%20%3A%2B1%3A%20because%20I%20guess%20it%20will%20come%20in%20an%20upcoming%20patch.%22%2C%20%22created_at%22%3A%20%222015-10-27T14%3A28%3A13Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/ConfigurableElement.cpp%3AL38-45%22%7D%2C%20%22Pull%202c8eda08af99554bf66de2148f1696324b1504c8%20parameter/ConfigurableDomain.cpp%206%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/296%23discussion_r43099858%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%3D%20default.%22%2C%20%22created_at%22%3A%20%222015-10-27T09%3A44%3A04Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22done%20in%20another%2C%20upcoming%20commit.%22%2C%20%22created_at%22%3A%20%222015-10-27T10%3A49%3A29Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-10-27T13%3A16%3A05Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/ConfigurableDomain.cpp%3AL42-53%22%7D%2C%20%22Pull%202c8eda08af99554bf66de2148f1696324b1504c8%20parameter/EnumValuePair.cpp%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/296%23discussion_r43099933%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%3D%20default%22%2C%20%22created_at%22%3A%20%222015-10-27T09%3A44%3A32Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22ditto%22%2C%20%22created_at%22%3A%20%222015-10-27T10%3A49%3A35Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-10-27T13%3A16%3A05Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/EnumValuePair.cpp%3AL33-40%22%7D%2C%20%22Pull%202c8eda08af99554bf66de2148f1696324b1504c8%20parameter/ParameterMgr.cpp%208%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/296%23discussion_r43130014%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22why%20did%20you%20not%20move%20those%20one%20%3F%22%2C%20%22created_at%22%3A%20%222015-10-27T14%3A40%3A10Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22Do%20not%20move%20them%2C%20you%20are%20right.%20The%20new%20here%20needs%20to%20be%20warded.%20%28BTW%20why%20is%20it%20not%20a%20member%20%3F%29%22%2C%20%22created_at%22%3A%20%222015-10-27T14%3A41%3A04Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-11-02T09%3A32%3A28Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/ParameterMgr.cpp%3AL320-330%22%7D%7D%2C%20%22approved%22%3A%20%7B%22https%3A//github.com/krocard%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/krocard'></a>

- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/296#issuecomment-151420999'>General Comment</a></b>
- <a href='https://github.com/dawagner'></a> @krocard @tcahuzax please review.
- <a href='https://github.com/dawagner'></a> @tcahuzax Please review.
- <a href='https://github.com/tcahuzax'></a> :-1:
- [x] <a href='#crh-comment-Pull 2c8eda08af99554bf66de2148f1696324b1504c8 parameter/ConfigurableDomain.cpp 6'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/296#discussion_r43099858'>File: parameter/ConfigurableDomain.cpp:L42-53</a></b>
- <a href='https://github.com/krocard'></a> = default.
- <a href='https://github.com/dawagner'></a> done in another, upcoming commit.
- [x] <a href='#crh-comment-Pull 2c8eda08af99554bf66de2148f1696324b1504c8 parameter/EnumValuePair.cpp 5'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/296#discussion_r43099933'>File: parameter/EnumValuePair.cpp:L33-40</a></b>
- <a href='https://github.com/krocard'></a> = default
- <a href='https://github.com/dawagner'></a> ditto
- [x] <a href='#crh-comment-Pull 2c8eda08af99554bf66de2148f1696324b1504c8 parameter/ConfigurableElement.cpp 5'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/296#discussion_r43128084'>File: parameter/ConfigurableElement.cpp:L38-45</a></b>
- <a href='https://github.com/krocard'></a> default :+1: because I guess it will come in an upcoming patch.
- [x] <a href='#crh-comment-Pull 2c8eda08af99554bf66de2148f1696324b1504c8 parameter/ParameterMgr.cpp 8'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/296#discussion_r43130014'>File: parameter/ParameterMgr.cpp:L320-330</a></b>
- <a href='https://github.com/krocard'></a> why did you not move those one ?
- <a href='https://github.com/krocard'></a> Do not move them, you are right. The new here needs to be warded. (BTW why is it not a member ?)

<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/296?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/296?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/296?approve=0'></a>&nbsp;<a href='https://github.com/01org/parameter-framework/pull/296'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by dawagner** [comment]
@krocard @tcahuzax please review.

File: parameter/ConfigurableDomain.cpp [hunk 1]
```diff
@@ -42,12 +42,11 @@
 
 using std::string;
 
-CConfigurableDomain::CConfigurableDomain() :
-    _bSequenceAware(false), _pLastAppliedConfiguration(NULL)
+CConfigurableDomain::CConfigurableDomain()
```
Comment (krocard) [inline parameter/ConfigurableDomain.cpp]: = default.

File: parameter/EnumValuePair.cpp [hunk 2]
```diff
@@ -33,7 +33,7 @@
 
 using std::string;
 
-CEnumValuePair::CEnumValuePair() : _iNumerical(0)
+CEnumValuePair::CEnumValuePair()
```
Comment (krocard) [inline parameter/EnumValuePair.cpp]: = default

File: parameter/ConfigurableDomain.cpp (hunk 1 above)
Comment (dawagner) [inline parameter/ConfigurableDomain.cpp]: done in another, upcoming commit.

File: parameter/EnumValuePair.cpp (hunk 2 above)
Comment (dawagner) [inline parameter/EnumValuePair.cpp]: ditto

Comment (dawagner) [inline parameter/EnumValuePair.cpp]: :+1:<a href='#crh-update-none'></a>

File: parameter/ConfigurableDomain.cpp (hunk 1 above)
Comment (dawagner) [inline parameter/ConfigurableDomain.cpp]: :+1:<a href='#crh-update-complete'></a>

File: parameter/ConfigurableElement.cpp [hunk 3]
```diff
@@ -38,7 +38,7 @@
 
 #define base CElement
 
-CConfigurableElement::CConfigurableElement(const std::string& strName) : base(strName), _offset(0)
+CConfigurableElement::CConfigurableElement(const std::string& strName) : base(strName)
```
Comment (krocard) [inline parameter/ConfigurableElement.cpp]: default :+1: because I guess it will cum in an upcomming patch.

File: parameter/ParameterMgr.cpp [hunk 4]
```diff
@@ -320,22 +320,10 @@ const CParameterMgr::SRemoteCommandParserItem CParameterMgr::gastRemoteCommandPa
 
 // Remote command parsers array Size
 CParameterMgr::CParameterMgr(const string& strConfigurationFilePath, log::ILogger& logger) :
-    _bTuningModeIsOn(false),
-    _bValueSpaceIsRaw(false),
-    _bOutputRawFormatIsHex(false),
-    _bAutoSyncOn(true),
     _pMainParameterBlackboard(new CParameterBlackboard),
```
Comment (krocard) [inline parameter/ParameterMgr.cpp]: why did you not move those one ?

Comment (krocard) [inline parameter/ParameterMgr.cpp]: Do not move them, you are right. The new here needs to be warded. (BTW why is it not a member ?)

**Comment by dawagner** [comment]
@tcahuzax Please review.

**Comment by tcahuzax** [comment]
:-1:

**Comment by krocard** [comment]
:shipit:

File: parameter/ParameterMgr.cpp (hunk 4 above)
Comment (krocard) [inline parameter/ParameterMgr.cpp]: :+1:

**Pull request merged**